src/components/LawProposals.vue

```diff
@@ -12,34 +15,68 @@ export default defineComponent({
 
   data() {
     return {
+      laws: [] as Law[],
       selectedLaw: undefined as LawId | undefined,
+      fadeingLaw: undefined as LawId | undefined,
     }
   },
 
-  props: {
-    proposedLaws: Array as PropType<Law[]>,
+  mounted() {
+    const e = this.$el as HTMLElement
+    e.style.setProperty("--transitiontime", TRANSITION_TIME + "s")
+  },
+
+  computed: {
+    ...mapGetters(["proposedLaws"]),
   },
 
   methods: {
-    select(lawId: LawId) {
+    select(lawId: LawId | undefined) {
       this.selectedLaw = lawId
     },
 
     accept(lawId: LawId) {
-      this.store.dispatch("acceptLaw", { lawId })
+      this.fadeingLaw = lawId
+      setTimeout(() => this.store.dispatch("acceptLaw", { lawId }), TRANSITION_TIME * 1000)
     },
 
     decline(lawId: LawId) {
-      this.store.dispatch("rejectLaw", { lawId })
+      this.fadeingLaw = lawId
+      setTimeout(() => this.store.dispatch("rejectLaw", { lawId }), TRANSITION_TIME * 1000)
     },
   },
+
+  watch: {
+    proposedLaws(newLaws: Law[]) {
+      const laws: Law[] = [...this.laws]
+      const addedLaws: Law[] = newLaws.filter(nl => !laws.some(ol => ol.id === nl.id))
+
+      for (var i = 0; i < laws.length; i++) {
+        if (newLaws.some(nl => nl.id === laws[i].id)) {
+          continue
+        }
+        if (addedLaws.length === 0) {
+          laws.splice(i)
+          continue
+        }
+        laws[i] = addedLaws.shift() as Law
+      }
+      laws.push(...addedLaws)
+      this.laws = laws
+    }
+  }
 })
 </script>
 
 <template>
   <div class="ProposedLaws">
-    <div v-for="(law, index) in proposedLaws" :key="index" class="Law" :class="{ selected: law.id === selectedLaw }">
-      <div @click="select(law.id)">
+    <div
+      v-for="(law, index) in laws"
+      :key="index"
+      class="Law"
+      :class="{ selected: law.id === selectedLaw, fading: law.id === fadeingLaw }"
+    >
+      <div @click="select(law.id)" @mouseenter="select(law.id)" @mouseleave="select(undefined)">
```

Comment:
Was ist denn der Grund dafür, den `hover` selbst zu implementieren?
Wenn wir es so machen, wäre es noch interessant, wie man die Selektion auch ohne Maus (auf Touch-Devices) wieder weg bekommt (war davon abgesehen vermutlich auch bisher ein Problem).

Reply:
Auf Desktop ist die Selektion bisher manchmal stehen geblieben. Das hat mich irritiert. So werden beide Arten über denselben State geführt. Wenn man auf eine Karte klickt löst der mouseleave die selektion wieder. Jetzt löst auch ein zweiter touch die selektion wieder. Alternativ könnte man den `hover` vielleicht nur auf dem Desktop und die Touch Selektion nur auf Mobile machen? Dafür reichen meine CSS Kenntnisse aber noch nicht. Da werden wir in Zukunft sicher sowieso noch rumfummeln. Vielleicht sollen die Gesetze ja auch gewischt werden...

Reply:
Da hast du recht.

Reply:
Ich konnte es nicht lassen und habe es mit dem "Checkbox Hack" hinbekommen, den ich bei der Gelegenheit gelernt habe. Mit `radio` geht das dann sogar auch als single-selection. Cool. Aber schon ein Hack...